addons/web/static/src/js/core/utils.js

```diff
@@ -204,14 +204,46 @@ var diacriticsMap = {
 };
 
 const patchMap = new WeakMap();
+/**
+ * @function serialize
+ * @param {any} object
+ * @returns {any}
+ */
+/**
+ * This object is where the special object types are defined. It can be further
+ * extended with more object types if needed. Object type definitions should have
+ * the following form:
+ * @property {class} Class the object constructor class (eg: Map, Date)
+ * @property {string} prefix a unique string used to deserialize the stringified
+ *      object (eg: "__MAP__")
+ * @property {serialize} serialize a function returning a serializable
+ *      form of the given object, like a string or an array.
+ */
+const UNSERIALIZABLE_TYPES = [
+    {
+        Class: Map,
+        prefix: "__MAP__",
+        serialize: map => [...map.entries()],
+    },
+    {
+        Class: Set,
+        prefix: "__SET__",
+        serialize: set => [...set.values()],
+    },
+];
 
+/**
+ * @function extractor
+ * @param {any} element
+ * @returns {any}
```

Comment:
deserves a description of the return value

Reply:
Done